### PR TITLE
Update MinHook to 1.3.3

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None

### Summary of the issue:
MinHook, the hooking library used to hook into processes, has been on 1.2.2 for years within NVDA. Version 1.3.3 is the most recent version.

### Description of how this pull request fixes the issue:
This updates MinHook to 1.3.3 and updates the associated sconscript file as well.

### Testing performed:
Try build ran without any observed issues, but wider testing would be appreciated.

### Known issues with pull request:
None

### CODE REVIEW

**Comment by josephsl** [comment]
Hi, any interesting things to look at based on changelog? Thanks.

**Comment by leonardder** [comment]
As it now stands, it looks like this pr doesn't allow my Edge to be started any more as soon as the try build is installed. :)

**Comment by feerrenrut** [comment]
> As it now stands, it looks like this pr doesn't allow my Edge to be started any more as soon as the try build is installed. :)

Can you confirm: this PR breaks Edge? Can you elaborate on what happens?

File: copying.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ All applicable licenses are included below.
 = GNU General Public License version 2 =
 Most of the source code for NVDA itself is available under this license.
 
-```
+``
```
Comment (feerrenrut) [inline copying.txt]: This should probably still be 3 ` characters

**Comment by leonardder** [comment]
[**@feerrenrut**](https://github.com/feerrenrut) commented on [9 apr. 2018 10:26 CEST](https://github.com/nvaccess/nvda/pull/8142#issuecomment-379674476 "2018-04-09T08:26:55Z - Replied by Github Reply Comments"):
> Can you confirm: this PR breaks Edge? Can you elaborate on what happens?

On one of my systems, as soon as I run this branch as an installed copy, Edge does seem to close around 2 seconds after I start it. When running as a portbale copy, this problem does not occur.

I have already tracked down that this is most likely cause by the more friendly error logging I tried to implement. Without these changes to error logging, A try build functions correctly. I'm currently investigating how to fix this.

**Comment by leonardder** [comment]
Error logging is now again working as expected.

File: nvdaHelper/remote/apiHook.cpp [hunk 2]
```diff
@@ -104,9 +104,9 @@ void* apiHook_hookFunction(const char* moduleName, const char* functionName, voi
 	}
 	LOG_DEBUG("requesting to hook function " << functionName << " at address 0X" << std::hex << realFunc << " in module " << moduleName << " at address 0X" << moduleHandle << " with  new function at address 0X" << newHookProc);
 	void* origFunc;
-	int res;
+	MH_STATUS res;
 	if((res=MH_CreateHook_fp(realFunc,newHookProc,&origFunc))!=MH_OK) {
```
Comment (feerrenrut) [inline nvdaHelper/remote/apiHook.cpp]: assignments done inside `if` blocks can easily become a source of error. I would much prefer to see this changed to:
```
MH_STATUS res = MH_CreateHook_fp(realFunc, newHookProc, &origFunc);
if( res != MH_OK) {
```
This gets rid of the declared but not initialised `res`, and reduces the logic inside the if statement.

**Review by feerrenrut — APPROVED** [review]

**Comment by leonardder** [comment]
Thanks for pointing this out. I've also addressed similar uses of res 
variables to maintain consistency.

**Review by feerrenrut — APPROVED** [review]

**Comment by kvark128** [comment]
On Windows 7 32bit with MinHook 1.3.3 there are great problems with UAC.
Not always, but often, after press Yes button in the UAC window, the window itself disappears, but nothing happens and the application does not start.
At the same time in task Manager hanging consent.exe process which does not terminated.
Replacing minHook.dll version 1.3.3 to 1.2.2 solves the problem for me.

**Comment by leonardder** [comment]
Thanks for your report. Could you please file a new issue for this, 
along with steps to reproduce?

**Pull request merged**